File: Algorithms/BinarySearch.py
data = [23,43,56,21,32,11,44,89,65,49]


# data sorting algorithm
def datasorting(data):
  
  for i in range(1,len(data)):
    key = data[i]
    j = i-1

    while(j>=0 and key<data[j]):
      
        data[j+1] = data[j]

        j-=1
      

    data[j+1]=key

# search algorithm
def BinarySearch(element,data):
  low =data[0]
  high = data[len(data)-1]
  medio = (high-low)/2
  if(element > data[len(data)-1] or element < data[0]):
    print("elemento no encontrado")
    return -1

  while(low<=high):
    
    if(element > medio):
      high =  medio+1
    elif(element < medio):
      high =  medio-1
    elif(element == medio):
      
      print(f"elemento encontrado")
    return element
  
  return -1


# test code
element = 11
datasorting(data)
print(data)
resultado = BinarySearch(element,data)

if(resultado != -1):
  print("element not found")

else:
  print("element has been found!")
  

  
    
      
    

  

  
    

  
    
  

File: Algorithms/BinarySearch.java
package Algorithms;

public class BinarySearch {
   

    
    public int binarysearch(int search,int[]data){
        int low = data[0];
        int high = data[data.length];
        int center = (high-low)/2;

        if(search == data[center]){
            System.out.println("Elemento encontrado");
            return search;
        }
        
        else if(search>data[center]){
            low = data[center+1];
            
        }

        else if(search<data[center]){
            high = data[center-1];

        }
        return -1;

        

    }
}
